src/components/CodeHighlight/LogColorizer.js

```diff
@@ -0,0 +1,194 @@
+import React, {
+  memo,
+  useState,
+  Fragment,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {
+  ButtonGroup,
+  Button,
+  Select,
+} from '@folio/stripes/components';
+
+import { LOG_COLORIZER } from '../../utils/constants';
+import { CodeHighlight } from './CodeHighlight';
+import * as Languages from './Languages';
+import * as Themes from './Themes';
+
+import css from './LogColorizer.css';
+
+const { LANGUAGES } = Languages;
+const {
+  THEMES,
+  themes,
+} = Themes;
+const { FILTER: { OPTIONS } } = LOG_COLORIZER;
+
+const filterOptions = [
+  {
+    id: OPTIONS.ALL,
+    caption: 'ui-data-import.logColorizer.filter.all',
+  }, {
+    id: OPTIONS.INFO,
+    caption: 'ui-data-import.logColorizer.filter.info',
+  }, {
+    id: OPTIONS.ERRORS,
+    caption: 'ui-data-import.logColorizer.filter.errors',
+  },
+];
+const themesPresent = [
+  {
+    id: 'coy',
+    caption: 'Coy',
+  }, {
+    id: 'stalker',
+    caption: 'Stalker',
+  },
+];
+
+export const LogColorizer = memo(props => {
+  const {
+    code,
+    language,
+    theme,
+    toolbar: {
+      visible,
+      message,
+      showThemes,
+    },
+  } = props;
+
+  const [currentFilter, setCurrentFilter] = useState(OPTIONS.ALL);
+  const [currentTheme, setCurrentTheme] = useState(theme);
+
+  const codePortion = Array.isArray(code) ? code : [code];
+  const hlTheme = themes[currentTheme];
+
+  const recordsCount = codePortion.length;
+  const errorsCount = codePortion.filter(item => JSON.stringify(item).includes('error')).length;
```

Comment:
Am I right that this code will count all records with any "error" text as errors even if they are not?
```suggestion
  const errorsCount = codePortion.filter(item => !!item.errorRecord).length;
```
Something like this would be more predictable and correct as `errorRecord` field is present only on error records.

Reply:
Done